src/hippopt/base/optimization_object.py

```diff
@@ -0,0 +1,64 @@
+from hippopt.common import (
+    Any,
+    ClassVar,
+    Type,
+    TypeVar,
+    Union,
+    abc,
+    copy,
+    cs,
+    dataclasses,
+    np,
+)
+
+TOptimizationObject = TypeVar("TOptimizationObject", bound="OptimizationObject")
+StorageType = Union[cs.MX, np.ndarray]
```

Comment:
Done in https://github.com/ami-iit/hippopt/pull/3/commits/ec4ad80fd41a5d1c07fb3abf8ed00a7fa86a10c1

Reply:
Note: for type checking with `isinstance`, it's necessary to use `np.ndarray` rather than the paramtererized generic `npt.NDArray`. Good catch that you didn't accept my suggestion in this sense :)

For all other usage instead, like for input/output types of methods/functions, please use `npt.NDArray`.

https://numpy.org/devdocs/reference/typing.html#numpy.typing.NDArray

Reply:
Ah, I see. Also it was not clear to me in the first place that ``npt`` was 
```
import numpy.typing as npt
```
Initially, I assumed it was a typo. 

Digression
This is what I meant by consistency. I am not sure if there are strict guidelines on which abbreviation to use. I mean, is ``npt`` universally accepted to be always ``numpy.typing``? If so, is there somewhere a documented list of abbreviations?

Reply:
The following are pretty established and there are IDEs that suggest them by default:

```python
import numpy as np
import numpy.typing as npt
import matplotlib.pyplot as plt
# ...
```

I pretty much agree with https://stackoverflow.com/a/53641923.